config/v1/types_authentication.go

```diff
@@ -159,3 +176,144 @@ const (
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+	// Issuer describes atributes of the OIDC token issuer
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Issuer TokenIssuer `json:"issuer"`
+
+	// ClaimMappings describes rules on how to transform information from an
+	// ID token into a cluster identity
+	ClaimMappings TokenClaimMappings `json:"claimMappings"`
+
+	// ClaimValidationRules are rules that are applied to validate token claims to authenticate users.
+	//
+	// +listType=atomic
+	ClaimValidationRules []TokenClaimValidationRule `json:"claimValidationRules,omitempty"`
+}
+
+// +kubebuilder:validation:MinLength=1
+type TokenAudience string
+
+type TokenIssuer struct {
+	// URL is the serving URL of the token issuer.
+	// Must use the https:// scheme.
+	//
+	// +kubebuilder:validation:Pattern=`^https:\/\/[^\s]`
+	// +kubebuilder:validation:Required
+	// +required
+	URL string `json:"issuerURL"`
+
+	// Audiences is an array of audiences that the token was issued for.
+	// Valid tokens must include at least one of these values in their
+	// "aud" claim.
+	// Must be set to exactly one value.
+	//
+	// +listType=set
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Audiences []TokenAudience `json:"audiences"`
+
+	// CertificateAuthority is a reference to a config map in the
+	// configuration namespace. The .data of the configMap must contain
+	// the "ca-bundle.crt" key.
+	// If unset, system trust is used instead.
+	CertificateAuthority ConfigMapNameReference `json:"issuerCertificateAuthority"`
+}
+
+type TokenClaimMappings struct {
+	// Username is a name of the claim that should be used to construct
+	// usernames for the cluster identity.
+	//
+	// Default value: "sub"
+	Username UsernameClaimMapping `json:"username,omitempty"`
+
+	// Groups is a name of the claim that should be used to construct
+	// groups for the cluster identity.
+	// The referenced claim must use array of strings values.
+	Groups PrefixedClaimMapping `json:"groups,omitempty"`
+}
+
+type TokenClaimMapping struct {
+	// Claim is a JWT token claim to be used in the mapping
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Claim string `json:"claim"`
+}
+
+type UsernameClaimMapping struct {
+	TokenClaimMapping `json:",inline"`
+
+	// By default, claims other than `email` will be prefixed with the issuer URL to
+	// prevent naming clashes with other plugins.
+	//
+	// Set to "-" to disable prefixing.
+	//
+	// Example:
+	//     (1) `prefix` is set to `myoidc:` and `claim` is set to "username".
+	//         If the JWT claim `username` contains value `userA`, the resulting
+	//         mapped value will be "myoidc:userA".
+	//     (2) `prefix` is set to `myoidc:` and `claim` is set to "email". If the
+	//         JWT `email` claim contains value "userA@myoidc.tld", the resulting
+	//         mapped value will be "userA@myoidc.tld".
+	Prefix string `json:"prefix"`
```

Comment:
playing with 


```go

type PrefixPolicy string

var(
  // NoOpinion let's the cluster assign prefixes.  If the username claim is email, there is no prefix
  // If the username claim is anything else, it is prefixed by the issuerURL
  NoOpinion PrefixPolicy = ""

  // NoPrefix means the username claim value will not have any  prefix
  NoPrefix PrefixPolicy = "NoPrefix"

  // Prefix means the prefix value must be specified.  It cannot be empty
  Prefix PrefixPolicy = "Prefix"
)

type UsernamePrefix struct{
  PrefixPolicy PrefixPolicy

  Prefix *Prefix
}

type Prefix struct{
  // required
  // minLength=1
  Prefix string
}

```

This structure would allow NoOpinion to match the kube-apiserver doc (I think).  And would make a selection here clearly indicate intent.